CLDC-1775 Bulk upload check your answers treatment

asmega opened the pull request:
# Context

- https://digital.dclg.gov.uk/jira/browse/CLDC-1775
- When checking answers for a bulk upload log, it get different treatment

# Changes

- When checking answers for a bulk upload log and the question is yet to be answered the copy is different and also coloured red
- `FilterService` class can now be instantiated. This is the start of having a better API for filtering and in my opinion beats having to include the `Modules::LogsFilter` which uses inheritance rather than composition

# Screenshots

![Screenshot 2023-02-01 at 16 56 55](https://user-images.githubusercontent.com/92580/216109890-6b0d2d9e-4d1a-45b2-8503-01e7ef5d65a0.png)

asmega committed:
CYA summary list tweaks for bulk upload

- When checking answers for a bulk upload related log the copy is
  tweaked and made red as per designs

asmega committed:
different CYA treatment when for bulk upload

kosiakkatrina commented on app/components/check_answers_summary_list_card_component.rb:
```diff
@@ -1,10 +1,12 @@
     @questions = questions
     @log = log
     @user = user
+    @bulk_upload = bulk_upload
```
Not against doing it like this, but checking if @log has bulk_upload_id might mean that we pass in less arguments here

asmega replied:
The behaviour would be different if it were implemented that way. Here the red version is only shown when the user is filtering logs via bulk upload. In your implementation it would always display the red version if a log was uploaded via bulk upload regardless if they were filtering by bulk upload. I will check in with designers to see which it is they are after

asmega replied:
Looks they were after what you suggested. I'll update this PR later to reflect that. I might keep the filtering stuff though and use it elsewhere as think that's worth keeping

asmega replied:
I've tweaked this now, so it's always red if the log was bulk uploaded and ignores any filtering the user has chosen

asmega committed:
bulk upload CYA missing answers always red

- whereas previosly this only happened if user was filtering logs via a
  bulk upload
